### DISCUSSION TITLE
Toggle to display and hide filter sidebar

### DISCUSSION THREAD

**Discussion by jpawlyn** [activeadmin/activeadmin#8435] [Ideas]
It looks like in activeadmin 4.0 the filter sidebar is always on show, sometimes on the right of the page and sometimes underneath.

We have quite a lot of columns showing for some of our resource index pages so we were thinking of re-introducing the filter toggle that exists in activeadmin 3.2. 

This could be a config option for users to choose which version of the filter sidebar they would like. 

Would this be of interest to anybody else? We want to understand if we should just build it in our app or if it would be a useful addition to the gem ?

**Comment by javierjulio** [comment]
@jpawlyn please provide examples/screenshots of what you are referring to. This didn't make sense to me. The Filters column shouldn't wrap underneath unless you are on a smaller screen (e.g. mobile device) which is intentional for it to be responsive. There is no config option other than to disable filters that I'm aware of at least. Are you referring to some plugin that added functionality in v3.x?

**Reply by jpawlyn** [reply]
Apologies @javierjulio, you are correct. We are using [arctic_admin](https://github.com/cprodhomme/arctic_admin?tab=readme-ov-file#arcticadmin) which explains the filter sidebar behaviour I described above.

Having removed the theme I can now see that the filter sidebar is static:

<img width="1417" alt="Screenshot 2024-08-21 at 10 27 05" src="https://github.com/user-attachments/assets/fb835547-00e8-4e1b-8e67-7026261d245d">
<br>
<br>
<br>
The issue we face is that with v4.0, the menu takes up space on the left and we need to scroll horizontally to see all the columns on some index pages (see below). Having the ability to toggle the display of the filter sidebar will be useful to us, so please let us know if you think this option (as non-default behaviour) may be of use in the gem.
<br>
<br>
<br>
<img width="1421" alt="Screenshot 2024-08-21 at 10 52 56" src="https://github.com/user-attachments/assets/1a5d2e21-9462-413c-9595-9f7a2399f212">

**Reply by javierjulio** [reply]
No worries at all. Ok thank you, this helps and makes sense now. For now, no, I do not want to change the default behavior here. This is meant to keep it simple on the library side but allow you to customize as you see fit since preferences will vary widely by giving you access to modifying the layout and sidebar partials if needed. This is by design and intentional. This could change in the future but it's a hard no for me. The expectation is that you are able to implement this with your own custom behavior and design since preferences here will vary widely. Thank you.

**Reply by jpawlyn** [reply]
@javierjulio Thanks for the explanation, I can appreciate the desire for simplicity. One thing though, the proposal above explicitly rules out changing the default behaviour.
